src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -3665,6 +3665,32 @@ public IAsyncResult BeginAccept(Socket? acceptSocket, int receiveSize, AsyncCall
             // Start the accept.
             DoBeginAccept(acceptSocket, receiveSize, asyncResult);
 
+            // Copy socket state if new socket was created when accepting into existing socket (Unix).
+            if (acceptSocket != null && acceptSocket != asyncResult.AcceptSocket)
+            {
+                asyncResult.AcceptSocket!._addressFamily = acceptSocket._addressFamily;
+                asyncResult.AcceptSocket!._cachedTaskEventArgs = acceptSocket._cachedTaskEventArgs;
+                asyncResult.AcceptSocket!._caches = acceptSocket._caches;
+                asyncResult.AcceptSocket!._closeTimeout = acceptSocket._closeTimeout;
+                asyncResult.AcceptSocket!._disposed = acceptSocket._disposed;
+                asyncResult.AcceptSocket!._handle = acceptSocket._handle;
+                asyncResult.AcceptSocket!._isConnected = acceptSocket._isConnected;
+                asyncResult.AcceptSocket!._isDisconnected = acceptSocket._isDisconnected;
+                asyncResult.AcceptSocket!._isListening = acceptSocket._isListening;
+                asyncResult.AcceptSocket!._lastReceiveHandle = acceptSocket._lastReceiveHandle;
+                asyncResult.AcceptSocket!._lastReceiveThread = acceptSocket._lastReceiveThread;
+                asyncResult.AcceptSocket!._lastReceiveTick = acceptSocket._lastReceiveTick;
+                asyncResult.AcceptSocket!._nonBlockingConnectInProgress = acceptSocket._nonBlockingConnectInProgress;
+                asyncResult.AcceptSocket!._nonBlockingConnectRightEndPoint = acceptSocket._nonBlockingConnectRightEndPoint;
+                asyncResult.AcceptSocket!._protocolType = acceptSocket._protocolType;
+                asyncResult.AcceptSocket!._receivingPacketInformation = acceptSocket._receivingPacketInformation;
+                asyncResult.AcceptSocket!._remoteEndPoint = acceptSocket._remoteEndPoint;
+                asyncResult.AcceptSocket!._rightEndPoint = acceptSocket._rightEndPoint;
+                asyncResult.AcceptSocket!._socketType = acceptSocket._socketType;
+                asyncResult.AcceptSocket!._willBlock = acceptSocket._willBlock;
+                asyncResult.AcceptSocket!._willBlockInternal = acceptSocket._willBlockInternal;
+            }
```

Comment:
A few thoughts:
- This should only be done on Unix
- The temporary socket needs to be disposed of in some way, but in a way that avoids it trying to close it's handle.  Might be best / easiest to do a GC.SuppressFinalize on it.
- Similarly the destination socket's handle probably needs to be closed prior to overwriting everything, but the semantics of that will need to be validated against the Windows behavior
- I'd need to double-check, but I believe there are multiple accept code paths, with this only covering one of them, e.g. I think there's another in SocketAsyncEventArgs.
- You might want to start by writing a bunch of tests that pass on Windows, and then getting the Unix implementation to pass them as well